scripts/salt-setup.sh

```diff
@@ -46,6 +46,37 @@ function apply_optional_states {
   fi
 }
 
+# Appends any prewarmed roles to a (non-salt) file, one role per line. salt-bootstrap takes care of setting up these roles during node setup.
+function add_single_role_for_cluster_salt {
+  local role=${1}
+  echo "${role}" >> /etc/salt/prewarmed_roles
+}
+
+function add_prewarmed_roles {
+  if [ "${INCLUDE_FLUENT}" == "Yes" ]; then
+    # Note: This will need to be changed if making changes to versions etc in the prewarmed image.
+    local fluent_prewarmed=${FLUENT_PREWARM_TAG}
+    echo "Adding ${fluent_prewarmed} to the list of roles for the final image"
+    add_single_role_for_cluster_salt ${fluent_prewarmed}
+  fi
+
+  if  [ "${STACK_TYPE}" == "CDH" -a ! -z "${CLUSTERMANAGER_VERSION}" -a ! -z "${CLUSTERMANAGER_BASEURL}" -a ! -z "${CLUSTERMANAGER_GPGKEY}" -a ! -z "${STACK_VERSION}" -a ! -z "${STACK_BASEURL}" -a ! -z "${STACK_REPOID}" ]; then
+    local prewarmed=${PREWARM_TAG}
+    echo "Adding ${prewarmed} to the list of roles for the final image"
+    add_single_role_for_cluster_salt ${prewarmed}
+  fi
+}
+
+function delete_unnecessary_files() {
+  # Salt (version as of this change: 3000.2) ends up taking a long time to load modules. vspere is an especially slow one taking 3 seconds.
+  # Salt does not seem to allow skipping module load ('disable_modules' only disables module usage, not module loading)
+  # So, deleting some modules which are not used, and tend to cause Exceptions / delays
+  find /opt/salt_3000.2/lib/python3.6/site-packages/salt/modules/ -name "*lxd*" -exec rm -f {} \;
+  find /opt/salt_3000.2/lib/python3.6/site-packages/salt/modules/ -name "*vsphere*" -exec rm -f {} \;
```

Comment:
Isn't vsphere removal caused the issues previously? Right, @oleewere

Reply:
Wil let Oliver comment, but I don't think it is related at all. See https://jira.cloudera.com/browse/CB-8321?focusedCommentId=3964516&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3964516
Every salt-minion log will have similar errors today, without any changes (instead of vspehere - it'll be some other module. The PID which runs into this error proceeds to perform additional operations, even in the log posted on CB-8321)

Reply:
on my side, on one of the node salt minion was not able to set the role (only one of it from 2 minion) and showed only this error in the log. could be a different reason. if this passes on AWS/Azure deployments, it's fine for me

Reply:
can we merge this patch without these deletions? as it will still need some validations (i guess) because of these changes - I would like to add a new metering related (prewarmed) role (using this code)

Reply:
see: https://github.com/hortonworks/cloudbreak-images/pull/441

Reply:
@oleewere  - again, the module deletion was not the reason for the revert of the previous commit. The fact that salt-bootstrap did not populate the file if it already existed is what caused the problems. Curious on wha testing was done on https://github.com/hortonworks/cloudbreak-images/pull/441 - since that is all that was pending here.